packages/ember-testing/lib/test.js

```diff
@@ -467,6 +469,111 @@ EmberApplication.reopen({
   }
 });
 
+if (isEnabled('ember-testing-instances')) {
+  EmberApplication.reopen({
+    /**
+      Builds a new instance of the application and registers test helpers on it.
+
+      Example:
+
+      ```javascript
+      var instance = run(App, 'buildTestInstance');
+      ```
+
+      @method buildTestInstance
+      @public
+    */
+    buildTestInstance() {
+      setupForTesting();
+
+      var instance = this.buildInstance();
+      this.runInstanceInitializers(instance);
+      instance.setupEventDispatcher();
+      instance.injectTestHelpers();
+
+      instance.overrideRouterLocation({ location: 'none' });
+
+      return instance;
+    }
+  });
+
+  EmberApplicationInstance.reopen({
+    /**
+      This property is a flag to easily determine if the object in question is
+      an ApplicationInstance or something else (e.g., an Application).
+
+      @property isApplicationInstance
+      @type {Boolean}
+      @default true
+      @public
+    */
+    isApplicationInstance: true,
+
+    /**
+      This property contains the testing helpers for the current instance. These
+      are created once `injectTestHelpers` has been called.
+
+      @property testHelpers
+      @type {Object}
+      @default null
+      @public
+    */
+    testHelpers: null,
+
+    /**
+      This injects test helpers into the `testHelpers` property bound to the
+      current context. Any callbacks registered with `onInjectHelpers` will be
+      called once the helpers have been injected.
+
+      In most cases, you shouldn't need to call this method directly, as using
+      `App.buildTestInstance` will create an instance and call this method on
+      it, and is the preferred way of testing with instances.
+
+      Example:
+      ```
+      instance.injectTestHelpers();
+      ```
+
+      @method injectTestHelpers
+      @public
+    */
+    injectTestHelpers() {
+      this.reopen({
+        willDestroy() {
+          this._super(...arguments);
+          this.removeTestHelpers();
+        }
+      });
+
+      this.testHelpers = {};
+      for (var name in helpers) {
+        this.testHelpers[name] = helper(this, name);
```

Comment:
I'm not really a fan of all this copying, but without a large refactoring i dont see a quick fix.

One idea, is rather then copying, the test frameworks gives us a TestHelpers object, something like

```js
this.testHelpers = new TestHelpers(applicationInstance);
// internally, this.testHelpers now has a reference to the app instance, and can just do the right thing internally.
```